frontend/src/metabase/entities/tables.js

```diff
@@ -167,6 +168,20 @@ const Tables = createEntity({
       }
 
       if (state[virtualQuestionId]) {
+        const virtualQuestion = state[virtualQuestionId];
+        if (
+          virtualQuestion.display_name !== card.name ||
```

Comment:
Maybe this long conditional can be named something simple and meaningful, and extracted to a constant.

Reply:
I believe this condition can be simply dropped because creating questions doesn't happen too often

Reply:
Looks pretty straightforward to my taste. As a side note, I think we should aim to extract entity reducers to their own files when they grow up to a certain size or at least switch to `redux-action's` `handleActions` helper

Reply:
> I believe this condition can be simply dropped because creating questions doesn't happen too often

@ranquild, it's an update action handler, so it'd probably get dispatched a bit more often. I'd vote for leaving the condition in as I can't really remember why there was an original check 🤔 Maybe we can do another PR on `master` removing it and let it bake there without backporting it into the release. @npfitz WDYT?

Reply:
@kulyk But there is a top-level `if` for `Questions.actionTypes.CREATE` already

Reply:
I was talking about dropping only `virtualQuestion.display_name !== card.name || ...` part

Reply:
> But there is a top-level if for Questions.actionTypes.CREATE already

Sorry, I'm not sure I get how it affects the update handler in this case 🤔

Reply:
I though `virtualQuestion.display_name !== card.name` will actually make sure we're going to update the name and fix the bug this PR is addressing, won't it? Like we're renaming a question and we want the "virtual table" name to get updated too

Reply:
I suspect @ranquild was referring to removing the entire conditional so that we call updateIn on any update

Reply:
Yep

Reply:
If this suggestion causes so much confusion, please ignore it